src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -166,6 +167,22 @@ public abstract class AbstractGitSCMSource extends SCMSource {
 
     private static final Logger LOGGER = Logger.getLogger(AbstractGitSCMSource.class.getName());
 
+    static Set<String> cacheEntries = new HashSet<>();
+
+    static {
```

Comment:
We need to be extremely cautious here. As a rule of thumb, whenever we touch any code that is common to the working of the whole plugin, we should be extra careful while placing any logic.